core/src/main/java/com/linecorp/armeria/common/SystemPropertyFlagsProvider.java

```diff
@@ -148,6 +149,22 @@ public Boolean useOpenSsl() {
         return getBoolean("useOpenSsl");
     }
 
+    @Override
+    public TlsEngineType tlsEngineType() {
+        final String strTlsEngineType = getNormalized("tlsEngineType");
+        if (strTlsEngineType == null) {
+            return null;
+        }
+        switch (strTlsEngineType) {
+            case "jdk":
+                return TlsEngineType.JDK;
+            case "openssl":
+                return TlsEngineType.OPENSSL;
+            default:
+                throw new IllegalArgumentException(String.format("%s isn't TlsEngineType", strTlsEngineType));
```

Comment:
It would be better to specify the correct values in the error message for easier debugging

```suggestion
                throw new IllegalArgumentException(String.format("%s isn't one of 'jdk' or 'openssl'", strTlsEngineType));
```

Reply:
Sure !